slides/10_SideEffects.html

```diff
@@ -0,0 +1,209 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8" />
+    <meta
+      name="viewport"
+      content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no"
+    />
+
+    <title>10 SideEffects | Essential React</title>
+
+    <link rel="icon" href="../img/zuehlke_logo.jpg" />
+    <link
+      rel="stylesheet"
+      href="https://cdn.jsdelivr.net/gh/webplatformz/zuehlke-reveal-package@aa0782bae9e270f4153d310e9f42bea172788e21/release/index.css"
+    />
+  </head>
+
+  <body>
+    <div class="reveal">
+      <div class="slides">
+        <section class="deck-slide">
+          <h1>
+            Essential React <br />
+            10 SideEffects
+          </h1>
+        </section>
+        <section>
+          <section>
+            <h2>useEffect</h2>
+            <p>What is it good for?</p>
+          </section>
+          <section>
+            <h3>Valid useCases</h3>
+            <ul>
+              <li>Subscribing to Events (window, ...)</li>
+              <li>Releasing resources to prevent a memory leak</li>
+              <li>Cancelling a network request/asynchronous operation</li>
+            </ul>
+          </section>
+          <section>
+            <pre><code class="hljs tsx" data-trim data-line-numbers="12-16|8-10|6,1-4|"><script type="text/template">
+              function getWindowDimensions() {
+                const { innerWidth: width, innerHeight: height } = window;
+                return { width, height};
+              }
+              export default function MyComponent() {
+                const [windowDimensions, setWindowDimensions] = useState(getWindowDimensions());
+              
+                const handleResize = useCallback(() => {
+                  setWindowDimensions(getWindowDimensions());
+                }, []);
+              
+                useEffect(() => {
+                  handleResize();
+                  window.addEventListener('resize', handleResize);
+                  return () => window.removeEventListener('resize', handleResize);
+                }, [handleResize]);
+              // ...
+              
+          </script></code></pre>
+          </section>
+
+          <section>
+            <h3>Cleanup is necessary for...</h3>
+            <ul>
+              <li>Unsubscribing from a pub-sub interface</li>
+              <li>Releasing resources to prevent a memory leak</li>
+              <li>Cancelling a network request/asynchronous operation</li>
+            </ul>
+          </section>
+          <section>
+            <h3>useEffect as last-resort</h3>
+            <ul>
+              <li>Don't use useEffect to update state on props or state</li>
```

Comment:
The Don'ts might need some more explanation (for the first I'm also not sure if I understand it -- do you mean don't mutate props? and what is meant re. state? a state setter can be called in useEffect)

Reply:
> https://beta.reactjs.org/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

Reply:
Ok, yes, that's good to bring in to avoid people overusing effects, quite interesting docs there. But I think then it's missing a word and therefore got me confused ;-) Shouldn't it be "Don't use useEffect to update state *based* on props or state"?

Also maybe we should add some more context. Generally, it's allowed to update state in useEffect (based on props and state). It's only when this is the only thing you want to do or when the state is redundant, that you should solve it simpler and not use an effect. At least that's how I understand the docs there.